hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/test/java/org/apache/hadoop/mapreduce/TestJobSubmissionFiles.java

```diff
@@ -139,4 +142,26 @@ public void testGetStagingDirWhenShortFileOwnerNameAndShortUserName()
     assertEquals(stagingPath,
         JobSubmissionFiles.getStagingDir(cluster, conf, user));
   }
+
+  @Test
+  public void testDirPermission() throws Exception {
+    Cluster cluster = mock(Cluster.class);
+    HdfsConfiguration conf = new HdfsConfiguration();
+    conf.set(CommonConfigurationKeys.FS_PERMISSIONS_UMASK_KEY, "700");
+    MiniDFSCluster dfsCluster = null;
+    try {
+        dfsCluster = new MiniDFSCluster.Builder(conf).numDataNodes(2).build();
```

Comment:
Please indent by 2 spaces instead of 4 to match Hadoop coding standards.

Once that's done, I think this will be ready to commit. Thanks!